README.tfdoc.hcl

```
@@ -0,0 +1,1265 @@
+              string. To compare against the query string, use a
+              `query_string` condition.
+            END
+
+            attribute "values" {
+              required    = true
+              type        = set(string)
+              description = <<-END
+                List of path patterns to match against the request URL. 
+              END
+            }
+          }
+
+          attribute "query_string" {
+            type        = set(query_string)
+            default     = []
+            description = <<-END
+              Query strings to match.
+            END
+
+            attribute "key" {
+              type        = string
+              description = <<-END
+                Query string key pattern to match. 
+              END
+            }
+
+            attribute "value" {
+              required    = true
+              type        = string
+              description = <<-END
+                Query string value pattern to match.
+              END
+            }
+          }
+
+          attribute "source_ip" {
+            type        = object(source_ip)
+            default     = []
+            description = <<-END
+              You can use both IPv4 and IPv6 addresses. Wildcards are not
+              supported. Condition is satisfied if the source IP address of the
+              request matches one of the CIDR blocks. Condition is not satisfied
+              by the addresses in the `X-Forwarded-For` header, use
+              `http_header` condition instead.
+            END
+
+            attribute "values" {
+              required    = true
+              type        = set(string)
+              description = <<-END
+                List of source IP CIDR notations to match.
+              END
+            }
+          }
+        }
+
+        attribute "authenticate_cognito" {
+          type        = object(authenticate_cognito)
+          default     = []
+          description = <<-END
+            Configuration block for using Amazon Cognito to authenticate users.
+            Specify only when `type` is `authenticate-cognito`.
+          END
+
+          attribute "user_pool_arn" {
+            required    = true
+            type        = string
+            description = <<-END
+              ARN of the Cognito user pool.
+            END
+          }
+
+          attribute "user_pool_domain" {
+            required    = true
+            type        = string
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user pool.
+            END
+          }
+
+          attribute "user_pool_client_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user
+              pool.
+            END
+          }
+
+          attribute "authentication_request_extra_params" {
+            type        = map(string)
```

```suggestion
            type        = list(authentication_request_extra_param)
```

This is a list of objects, please add the attributes of the object also

---

The schema says it is:
```
"authentication_request_extra_params": {
                                                        "type": [
                                                            "map",
                                                            "string"
                                                        ],
                                                        "description_kind": "plain",
                                                        "optional": true
                                                    },
```

I can see the terraform registry documentation differs, I'm not sure which one to go with.

---

let's go with provider schema